src/TrackingCollectionTests/TrackingCollectionTests.cs

```diff
@@ -429,8 +429,8 @@ public void OrderByMatchesOriginalOrder()
         var count = 0;
         var total = 1000;
 
-        var list1 = new List<Thing>(Enumerable.Range(1, total).Select(i => GetThing(i, i, total - i, "Run 1")).ToList());
-        var list2 = new List<Thing>(Enumerable.Range(1, total).Select(i => GetThing(i, i, total - i, "Run 2")).ToList());
+        var list1 = new List<Thing>(Enumerable.Range(1, total).Select(i => GetThing(i, i, i, "Run 1")).ToList());
```

Comment:
This test is testing whether `OrderBy` will correctly retain the order of the collection if all items added to it are in the same order as the `OrderBy` comparer. You're changing the order here, which changes the conditions of the test (by forcing the collection to actually reorder). Not that it's going to fail the test, but that's not what this test is supposed to do, this test is exercising the code path where no re-ordering is required.